### PR TITLE
Add support for GHC 9.6 & update CI

### PR DESCRIPTION
**Pull request by wismill**

### CODE REVIEW

**Comment by wismill** [comment]
I updated CI, but currently we cannot test GHC 9.6 because [ghcup is not updated](https://github.com/haskell/actions/issues/37#issuecomment-907861632).

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -102,13 +102,18 @@ jobs:
...
 
-          - name: 9.4.2
-            ghc_version: 9.4.2
+          - name: 9.4.4
+            ghc_version: 9.4.4
+            runner: ubuntu-latest
+            cabal_version: 3.8.1.0
+
+          - name: 9.6.0
+            ghc_version: 9.6.0
```
Comment (harendra-kumar) [inline .github/workflows/haskell.yml:116]: We have a working ghc head version CI in streamly. You can try adding it like this: https://github.com/composewell/streamly/blob/ab28951742e0a423a27762a5c9cccaf49745c253/.github/workflows/haskell.yml#L193

**Review by harendra-kumar — APPROVED** [review]

**Comment by wismill** [comment]
GHC head added

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (wismill) [reply .github/workflows/haskell.yml]: Done

**Comment by wismill** [comment]
So my PR has been accepted to use ghcup 0.1.19 in actions. I am trying to figure out how to install GHC 9.6.0. It works on my computer.

**Comment by wismill** [comment]
Waiting for new PR to add support for ghcup release channels in actions.

**Comment by wismill** [comment]
CI is finally successful for GHC 9.6.0!

**Comment by wismill** [comment]
Revisions published on Hackage to allow to build with `base` < 4.19.

**Comment by harendra-kumar** [comment]
Ship it!

**Pull request merged**